design-proposals/sriov-interfaces-mapping.md

```diff
@@ -0,0 +1,694 @@
+# Overview
+Accurately associate VMs SR-IOV interfaces with the underlying SR-IOV Virtual Functions (VF) devices (PCI) allocated by the sriov-network-device-plugin and sriov-cni.
```

Comment:
Note that for secondary SR-IOV networks Kubevirt VMs consume VFs with vfio-pci driver and not "network-device  mode" VFs in order to pass through them to the guest
https://kubevirt.io/user-guide/virtual_machines/interfaces_and_networks/#how-to-expose-sr-iov-vfs-to-kubevirt.

In any case, if a CNI reports the PCI address of the VF according to the NPWG docs,
it will be reflected in the networks-status annotations and the mapping will be correct (as mentioned in the doc, if the network-to-pci info is missing Kubevirt falls back to the legacy mapping method)

As for the primary network, as long as the VF is in "network device mode" Kubevirt will wire it up transparently.
Thanks for the heads up, I will add changes to reflect that it's also supported.

Reply:
Following the discussion at https://github.com/kubevirt/kubevirt/pull/8226#discussion_r940012523, no need to reflect the mentioned above.